File: ssm-sdk-client-sign/src/main/kotlin/ssm/client/sign/utils/FileUtils.java
package ssm.client.sign.utils;

import java.io.File;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.Reader;
import java.net.MalformedURLException;
import java.net.URL;

import com.google.common.io.Resources;

public class FileUtils {

    public static final String FILE = "file:";

    public static File getFile(String filename) throws MalformedURLException {
        URL url = getUrl(filename);
        return new File(url.getFile());
    }

    public static URL getUrl(String filename) throws MalformedURLException {
        if(filename.startsWith(FILE)) {
            return new URL(filename);
        }
        return Resources.getResource(filename);
    }

    public static Reader getReader(String filename) throws IOException {
        URL url = getUrl(filename);
        return new InputStreamReader(url.openStream());
    }
}

File: ssm-sdk-client-sign/src/main/kotlin/ssm/client/sign/utils/FileUtils.kt
package ssm.client.sign.utils

import com.google.common.io.Resources
import java.io.File
import java.io.IOException
import java.io.InputStreamReader
import java.io.Reader
import java.net.MalformedURLException
import java.net.URL

class FileUtils {
    companion object {
        const val FILE = "file:"

        @Throws(MalformedURLException::class)
        fun getFile(filename: String): File {
            val url = getUrl(filename)
            return File(url.file)
        }

        @Throws(MalformedURLException::class)
        fun getUrl(filename: String): URL {
            return if (filename.startsWith(FILE)) {
                URL(filename)
            } else Resources.getResource(filename)
        }

        @Throws(IOException::class)
        fun getReader(filename: String): Reader {
            val url = getUrl(filename)
            return InputStreamReader(url.openStream())
        }
    }
}